src/msac.rs

```diff
@@ -149,10 +149,49 @@ impl Default for Rav1dMsacDSPContext {
 
 pub type EcWin = usize;
 
+/// # Safety
+///
+/// [`Self`] must be the first field of [`MsacAsmContext`] for asm layout purposes.
+/// And [`Self::pos`] and [`Self::end`] must be either [`ptr::null`],
+/// or [`Self::pos`] must point into (or the end of) [`MsacContext::data`],
```

Comment:
I think it's important to point out that it must be the containing `MsacContext`, not some unrelated buffer.

Reply:
Okay I'll point that out.

Reply:
Fixed in [`1ec6eab..6856860`](https://github.com/memorysafety/rav1d/compare/1ec6eab4e00770b6022a66294f7f3c91b3967257..6856860f24d841e011e8a69508e417699b1042c0).